reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/TimedScheduler.java

```diff
@@ -0,0 +1,274 @@
+		this.submittedDirect.increment();
+		return delegate.schedule(wrap(task));
+	}
+
+	@Override
+	public Disposable schedule(Runnable task, long delay, TimeUnit unit) {
+		this.submittedDelayed.increment();
+		return delegate.schedule(wrap(task), delay, unit);
+	}
+
+	@Override
+	public Disposable schedulePeriodically(Runnable task, long initialDelay, long period, TimeUnit unit) {
+		this.submittedPeriodicallyInitial.increment();
+		return delegate.schedulePeriodically(wrapPeriodical(task), initialDelay, period, unit);
+	}
+
+	@Override
+	public Worker createWorker() {
+		return new TimedWorker(this, delegate.createWorker());
+	}
+
+	@Override
+	public boolean isDisposed() {
+		return delegate.isDisposed();
+	}
+
+	@Override
+	public long now(TimeUnit unit) {
+		return delegate.now(unit);
+	}
+
+	@Override
+	public void dispose() {
+		delegate.dispose();
+	}
+
+	@Override
+	public void start() {
+		delegate.start();
+	}
+
+	static final class TimedWorker implements Worker {
+
+		final TimedScheduler parent;
+		final Worker      delegate;
+
+		TimedWorker(TimedScheduler parent, Worker delegate) {
+			this.parent = parent;
+			this.delegate = delegate;
+		}
+
+		@Override
+		public void dispose() {
+			delegate.dispose();
+		}
+
+		@Override
+		public boolean isDisposed() {
+			return delegate.isDisposed();
+		}
+
+		@Override
+		public Disposable schedule(Runnable task) {
+			parent.submittedDirect.increment();
+			return delegate.schedule(parent.wrap(task));
+		}
+
+		@Override
+		public Disposable schedule(Runnable task, long delay, TimeUnit unit) {
+			parent.submittedDelayed.increment();
+			return delegate.schedule(parent.wrap(task), delay, unit);
+		}
+
+		@Override
+		public Disposable schedulePeriodically(Runnable task, long initialDelay, long period, TimeUnit unit) {
+			parent.submittedPeriodicallyInitial.increment();
+			return delegate.schedulePeriodically(parent.wrapPeriodical(task), initialDelay, period, unit);
+		}
+	}
+
+	static final class TimedRunnable implements Runnable {
+
+		final MeterRegistry registry;
+		final TimedScheduler parent;
+		final Runnable task;
+
+		final LongTaskTimer.Sample pendingSample;
+
+		boolean isRerun;
+
+		TimedRunnable(MeterRegistry registry, TimedScheduler parent, Runnable task) {
+			this(registry, parent, task, false);
+		}
+
+		TimedRunnable(MeterRegistry registry, TimedScheduler parent, Runnable task, boolean periodically) {
+			this.registry = registry;
+			this.parent = parent;
+			this.task = task;
+
+			if (periodically) {
+				this.pendingSample = null;
+			}
+			else {
+				this.pendingSample = parent.pendingTasks.start();
+			}
+			this.isRerun = false; //will be ignored if not periodical
+		}
+
+		@Override
+		public void run() {
+			if (this.pendingSample != null) {
+				//NOT periodical
+				this.pendingSample.stop();
+			}
+			else {
+				if (!isRerun) {
+					this.isRerun = true;
+				}
+				else {
+					parent.submittedPeriodicallyIteration.increment();
```

Comment:
The implementation indicates that the description of the name of this meter is inaccurate.

Reply:
fixed